modules/node-shared/src/main/scala/org/tessellation/node/shared/http/routes/MetagraphMessageRoutes.scala

```diff
@@ -0,0 +1,57 @@
+package org.tessellation.node.shared.http.routes
+
+import cats.Show
+import cats.data.Validated.{Invalid, Valid}
+import cats.effect.Async
+import cats.syntax.flatMap._
+import cats.syntax.functor._
+import cats.syntax.show._
+
+import org.tessellation.node.shared.infrastructure.snapshot.services.MetagraphMessageService
+import org.tessellation.node.shared.infrastructure.snapshot.services.MetagraphMessageService.MetagraphMessageServiceErrorOr
+import org.tessellation.routes.internal._
+import org.tessellation.schema.http.{ErrorCause, ErrorResponse}
+import org.tessellation.schema.metagraphMessage.{AdminFeeKeyMessage, StakingMessage}
+import org.tessellation.security.{Hasher, SecurityProvider}
+
+import eu.timepit.refined.auto._
+import org.http4s.circe.CirceEntityCodec.{circeEntityDecoder, circeEntityEncoder}
+import org.http4s.dsl.Http4sDsl
+import org.http4s.{HttpRoutes, Response}
+import org.typelevel.log4cats.slf4j.Slf4jLogger
+
+class MetagraphMessageRoutes[F[_]: Async: SecurityProvider: Hasher](
+  service: MetagraphMessageService[F]
+) extends Http4sDsl[F]
+    with PublicRoutes[F] {
+
+  private val logger = Slf4jLogger.getLoggerFromName("MetagraphMessageRoutes")
+
+  protected val prefixPath: InternalUrlPrefix = "/currency"
+
+  protected val public: HttpRoutes[F] = HttpRoutes.of[F] {
+    case req @ POST -> Root / "signing" / "owner" =>
```

Comment:
maybe `owner/sign` and `staking/sign`? also, looks like it does not sign it here, so maybe just `owner` and `staking` without `sign` or `signing`?

Reply:
Changed to single endpoint, `/currency/message`